src/pages/Search/SearchPageNarrow.tsx

```diff
@@ -119,6 +121,17 @@ function SearchPageNarrow({queryJSON, headerButtonsOptions, searchResults, isMob
         }
     }, []);
 
+    const footerData = useMemo(() => {
+        const selectedTransactionsKeys = Object.keys(selectedTransactions);
+        const shouldUseClientTotal = selectedTransactionsKeys.length > 0 && !areAllMatchingItemsSelected;
+
+        const currency = metadata?.currency;
+        const count = shouldUseClientTotal ? selectedTransactionsKeys.length : metadata?.count;
+        const total = shouldUseClientTotal ? Object.values(selectedTransactions).reduce((acc, transaction) => acc - transaction.convertedAmount, 0) : metadata?.total;
+
+        return {count, total, currency};
+    }, [areAllMatchingItemsSelected, metadata?.count, metadata?.currency, metadata?.total, selectedTransactions]);
+
```

Comment:
We have this logic duplicated here. Can we create a util function and just call that in both places please?